src/markov.rs

```diff
@@ -30,99 +31,108 @@ impl Key for Markov {
 impl Markov {
     pub fn new() -> Markov {
         Markov {
-            assocs: HashMap::new(),
-            start: HashMap::new(),
+            word_map: HashMap::new(),
+            starting_words: HashMap::new(),
         }
     }
 
     pub fn parse(&mut self, string: &str) {
-        // TODO: sanitize string of stuff like URLs
+        let url_regex = Regex::new(r"[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_\+.~#?&//=]*)").unwrap();
+        let string = url_regex.replace_all(string, "");
```

Comment:
This stuff needs to be in a separate private function.

Also, since the regex is constant, it should be static, since it doesn't change. As it is right now, you have to recreate it _for every message_. Slow.

Using the `lazy_static` package. Something like:

```rust
// In main.rs
#[macro_use]
extern crate lazy_static;

// In this file
lazy_static! {
    static ref URL_REGEX: Regex::new(...).unwrap();
}
```